### ISSUE TITLE
Add support for the stack, priority and resolution

### ISSUE THREAD

**Issue by meindratheal** [meindratheal/CavernsAndCards#1]
This is the groundwork for future card types and effects. The current implementation only allows the turn player to play 0 or 1 cards on their turn, and players can't play cards at all when it's not their turn. The ability to play multiple cards in a turn, or to be able to respond to other players' plays would make the game more interesting.

The terms Stack, Priority and Resolution are from MTG's turn structure.
Resolution is the execution of the effects of a card. For example, Attack cards will (attempt to) deal damage to the enemy player upon resolution.
The Stack is a holding location for cards that have been played but have not yet been resolved.
Priority is what determines the order in which players may play cards onto the stack. This also prevents stalls where each player waits for the other to play a card first.

The rules are as follows:

1. A phase in a turn begins. Currently, this only occurs within the Battle phase.
2. The turn player gets Priority.
3. The player with priority may play a card, subject to any restrictions on that card. For example, all current cards at the time of writing can only be played on your turn, and only one card card can be played per turn. This card is added to the top of the stack.
4. If the player with priority plays a card, return to 2 (the turn player always gets first chance to respond, whoever played the last card - including themselves). Otherwise, if the next player in turn order from the person with priority is not the turn player, this player gets priority and return to 3. Otherwise, continue to 5.
5. Every player has passed priority without playing a card. If the stack has at least one card, continue to 6. Otherwise, end the phase and exit this flowchart.
6. Remove the top card of the stack and resolve it (applying any effects). For example the only card type available at the time of writing is the Attack card type, which has a chance to deal damage to the enemy player. Then return to 2.

Note that in MTG the stack is used for (almost) all cards and abilities, and cards become a separate entity on the stack. Currently we are not looking to emulate this, but it may be required to implement other effects later in the game, particularly delayed ones.

In plain English:
- Turn phase starts. Turn player gets priority
- If anyone plays a card, priority returns to the turn player
- If everyone opts not to play a card, the top card is resolved and priority returns to the turn player. If there was no card left in the stack, instead the phase ends.
